### DISCUSSION TITLE
Why does the gridpoints endpoint return durations?

### DISCUSSION THREAD

**Discussion by robbieh** [weather-gov/api#613] [General]
Why does the API return forecasts with a duration like "2023-01-17T10:00:00+00:00/PT3H" instead of just a series of three values? I understand that's a little more efficient to transmit, but it is much more painful to work with on the receiving end. Instead of being able to look up the forecast for a given hour, I have to parse all the durations and figure out which is the right one.

**Comment by dwhitemv25** [comment]
It's an artifact of how the data is exported from the National Digital Forecast Database (NDFD), but yes, it's to reduce the data transmitted. (It's presented the same way in the NDFD XML and REST APIs.) 

The values are ISO 8601 time intervals, so if you have a library that can search time ranges then that would save some work. Alternately you can just expand the intervals into hourly steps, or whatever your software uses.
